src/utils/colors.js

```diff
@@ -73,3 +73,114 @@ export const GREY_90 = '#0c0c0d';
+  +unselectedFillStyle: string,
+  +selectedTextColor: string,
+  +gravity: number,
+|};
+
+/**
+ * Map a color name, which comes from Gecko, into a CSS style color. These colors cannot
+ * be changed without considering the values coming from Gecko, and from old profiles
+ * that already have their category colors saved into the profile.
+ *
+ * Category color names come from:
+ * https://searchfox.org/mozilla-central/rev/0b8ed772d24605d7cb44c1af6d59e4ca023bd5f5/tools/profiler/core/platform.cpp#1593-1627
+ */
+export function mapCategoryColorNameToStyles(colorName: string): ColorStyles {
+  switch (colorName) {
+    case 'transparent':
+      return {
+        selectedFillStyle: 'transparent',
+        unselectedFillStyle: 'transparent',
+        selectedTextColor: '#000',
+        gravity: 0,
+      };
+    case 'purple':
+      return {
+        selectedFillStyle: PURPLE_70,
+        // Colors are assumed to have the form #RRGGBB, so concatenating 2 more digits to
+        // the end defines the transparency #RRGGBBAA.
+        unselectedFillStyle: PURPLE_70 + '60',
+        selectedTextColor: '#fff',
+        gravity: 5,
+      };
+    case 'green':
+      return {
+        selectedFillStyle: GREEN_60,
+        unselectedFillStyle: GREEN_60 + '60',
+        selectedTextColor: '#fff',
+        gravity: 4,
+      };
+    case 'orange':
+      return {
+        selectedFillStyle: ORANGE_50,
+        unselectedFillStyle: ORANGE_50 + '60',
+        selectedTextColor: '#fff',
+        gravity: 2,
+      };
+    case 'yellow':
+      return {
+        selectedFillStyle: YELLOW_50,
+        unselectedFillStyle: YELLOW_50 + '60',
+        selectedTextColor: '#000',
+        gravity: 6,
+      };
+    case 'lightblue':
+      return {
+        selectedFillStyle: BLUE_40,
+        unselectedFillStyle: BLUE_40 + '60',
+        selectedTextColor: '#000',
+        gravity: 1,
+      };
+    case 'grey':
+      return {
+        selectedFillStyle: GREY_30,
+        unselectedFillStyle: GREY_30 + '60',
+        selectedTextColor: '#000',
+        gravity: 8,
+      };
+    case 'blue':
+      return {
+        selectedFillStyle: BLUE_60,
+        unselectedFillStyle: BLUE_60 + '60',
+        selectedTextColor: '#fff',
+        gravity: 3,
+      };
+    case 'brown':
+      return {
+        selectedFillStyle: MAGENTA_60,
+        unselectedFillStyle: MAGENTA_60 + '60',
+        selectedTextColor: '#fff',
+        gravity: 7,
+      };
+    default:
+      console.error(
+        'Unknown color name encountered. Consider updating this code to handle it.'
+      );
+      return {
+        selectedFillStyle: GREY_30,
+        unselectedFillStyle: GREY_30 + '60',
+        selectedTextColor: '#000',
+        gravity: 8,
+      };
+  }
+}
+
+/**
+ * This function tweaks the colors for the stack chart, but re-uses most
+ * of the logic from `mapCategoryColorNameToStyles`.
+ */
+export function mapStackChartColorNameToStyles(colorName: string): ColorStyles {
```

Comment:
Nit: We're still mapping a category color in this function, but I think this name makes it sound like we're mapping from a `StackChartColorName`. How about renaming this function to something like `mapCategoryColorNameToStackChartStyles` instead?